<launch>
    <!-- set to value="gdbserver localhost:10000" for remote debugging -->
    <arg name="launch_prefix" default="" />

    <!-- configure camera input -->
    <arg name="camera_name" default="/bebop" />
    <arg name="image_topic" default="image_raw" />
    <arg name="queue_size" default="1" />

    <!-- apriltag_ros continuous detection node -->
    <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="apriltag_ros_continuous_node" clear_params="true" output="screen" launch-prefix="$(arg launch_prefix)">
        <!-- Remap topics from those used in code to those on the ROS network -->
        <remap from="image_rect" to="$(arg camera_name)/$(arg image_topic)" />
        <remap from="camera_info" to="$(arg camera_name)/camera_info" />

        <param name="publish_tag_detections_image" type="bool" value="true" /><!-- default: false -->
        <param name="queue_size" type="int" value="$(arg queue_size)" />

        <!-- load parameters (incl. tag family, tags, etc.) -->
        <rosparam command="load" file="$(find bebop2_controller)/config/settings.yaml"/>
        <rosparam command="load" file="$(find bebop2_controller)/config/tags.yaml"/>
    </node>


    <node pkg="rviz" type="rviz" name="tagView" args="-d  $(find bebop2_controller)/config/tag.rviz" />

    <node pkg="bebop2_controller" type="bebop2_controller_node" name="bebop2_controller" output="screen" >
        <rosparam command="load" file="$(find bebop2_controller)/config/param.yaml" />
    </node>

    <!--    static frames-->
    <include file="$(find bebop2_controller)/launch/static_frames.launch" />



</launch>